### ISSUE TITLE
Configure.pl should detect compilers that won’t work

### ISSUE THREAD

**Issue by ugexe** [rakudo/rakudo#3352]
MoarVM is no longer restricted to C89. However, this means certain windows configurations using something older than VS2017 will no longer compile Rakudo. Since many appveyor setups use VS2012 it would be a good idea for the build system to give an error noting the incompatible compiler instead of trying to derive that knowledge from the appveyor output ( https://ci.appveyor.com/project/ugexe/zef/builds/29531256 )

**Comment by vrurg** [comment]
@ugexe can you provide a sample of version report from the compiler?

**Comment by ugexe** [comment]
On VS2017 a VM gives the following result as the first line when running `cl /?`

```
C:\Users\ugexe\.rakudobrew>cl /?
Microsoft (R) C/C++ Optimizing Compiler Version 19.12.25831 for x64
...
```

**Comment by vrurg** [comment]
This one I also have. I wonder what 2012 prints to decide how to handle it correctly.

**Comment by vrurg** [comment]
I don't think we want to limit ourselves to 2017 exclusively. :)

**Comment by ugexe** [comment]
VS2012 on appveyor:
```
cmd /c "cl.exe"
Microsoft (R) C/C++ Optimizing Compiler Version 16.00.40219.01 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.
```

**Comment by vrurg** [comment]
BTW, I'm adding the check to nqp-configure, so both NQP and Rakudo will be bailing out on anything less than 19.0. But MoarVM needs separate handling which is out of my hands.

**Comment by vrurg** [comment]
Closed with merge of #3359
